### ISSUE TITLE
html output for markdown: different output when using '# Header {#mainpage}'  and 'Header {#mainpage}\n===='

### ISSUE THREAD

**Issue by AndyHeather** [doxygen/doxygen#7043]
When creating html docs from markdown input, the following works:
```
TEST   {#mainpage}
====

abc def ghi jkl mno pqr stu ...
```
and produces: ![screenshot](https://user-images.githubusercontent.com/11627707/59114786-dcb05780-893f-11e9-8f27-b7ecc88778b3.png)

whereas the following:
```
# TEST   {#mainpage}

abc def ghi jkl mno pqr stu ...
```
produces: ![screenshot](https://user-images.githubusercontent.com/11627707/59114851-fa7dbc80-893f-11e9-85ef-1bf9b3f89d8f.png)

Is this expected behaviour?

Test case : [Markdown-section-header.tar.gz](https://github.com/doxygen/doxygen/files/3266537/Markdown-section-header.tar.gz)

Tested using current master branch at commit e773e6cb

**Comment by albert-github** [comment]
No I don't think there should be the difference.

I've just pushed a proposed patch, pull request #7045

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
